monolake-services/src/http/handlers/route.rs

```diff
@@ -254,26 +256,42 @@ where
         &self,
         old: Option<&Self::Service>,
     ) -> Result<Self::Service, Self::Error> {
-        let mut router: Router<RouteConfig> = Router::new();
-        for route in self.routes.iter() {
-            router.insert(&route.path, route.clone())?;
-            if route.upstreams.is_empty() {
-                return Err(RoutingFactoryError::EmptyUpstream);
-            }
-        }
-        Ok(RewriteAndRouteHandler {
-            inner: self
-                .inner
-                .make_via_ref(old.map(|o| &o.inner))
-                .await
-                .map_err(RoutingFactoryError::Inner)?,
-            router,
-        })
+        let router = Router::new_from_iter(self.routes.clone())?;
+        Ok(HttpErrorResponder(SvcRoute {
+            svc: RewriteHandler {
+                inner: self
+                    .inner
+                    .make_via_ref(old.map(|o| &o.0.svc.inner))
+                    .await
+                    .map_err(RoutingFactoryError::Inner)?,
+            },
+            selector: router,
+            selector_mapper: PathExtractor,
+        }))
     }
 }
 
-const fn default_weight() -> u16 {
-    1
+pub struct HttpErrorResponder<T>(pub T);
```

Comment:
Can we move this to the http module, guessing other handlers might want to use this as well?